### PR TITLE
Sprint3

### PR DESCRIPTION
**Pull request by NoxPardalis**

### CODE REVIEW

File: site/static/scripts/ui/domInteraction.js [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+    var serversListElem = document.getElementById("serverNamesList");
+    removeChildren(serversListElem);
+
+    for (var name in servers){
+        // Create element for each device
+        var elem = document.createElement("li");
+        elem.className = "device_row";
+        elem.id = name;
+
+        // Creates button to delete server
```
Comment (Feikoritsema) [inline site/static/scripts/ui/domInteraction.js]: I think this button could be improved, it's rather big and doesn't really fit there.

File: site/static/scripts/ui/domInteraction.js [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+            case "float":
+                var slideContainer = document.createElement("div");
+                slideContainer.className = "slidecontainer";
+                var input = document.createElement("input");
+                input.type = "range";
+                input.min = 0;
+                input.max = 100;
+                input.step = 10;
+                input.name = deviceId;
+                input.value = activator.state * 100; // TODO Based on object
```
Comment (Feikoritsema) [inline site/static/scripts/ui/domInteraction.js]: TODOs in the release isn't really appropriate.

File: site/static/scripts/ui/domInteraction.js [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+    var serverName = document.getElementById("serverNameInput").value;
+    var serverAddress = document.getElementById("serverAddressInput").value;
+    addUserServer(firebase, firebase.auth().currentUser, serverName.toString(), serverAddress.toString());
+}
+
+function submitRenameDevice() {
+    var deviceId = this.parentNode.id;
+    var newName = prompt("Enter name:","Device");
+
+    // Todo: Deal with empty input
```
Comment (Feikoritsema) [inline site/static/scripts/ui/domInteraction.js]: Same, but good that you mention it, it's necessary.

File: site/static/scripts/ui/domInteraction.js [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+
+/**
+ * Generates the html to view a device with a particular deviceName and for its
+ * associated activators. deviceId is passed so that we may interact with the
+ * activators and know their device. This will be refactored.
+ * @param {} deviceName
+ * @param {} deviceId
+ * @param {} activators
+ */
+function populateDeviceActivators(deviceName, deviceId, activators) {
```
Comment (Feikoritsema) [inline site/static/scripts/ui/domInteraction.js]: Really huge function, might want to split it up.

**Comment by PapaOscar90** [comment]
This sprint, the team has gone ahead and entirely re-written the code base. We have switched from the old purely javascript running on flask, to a robust and well supported frontend and self hosted backend system. Though this is still in separate feature branches, we intend to have everything completed by the end of this sprint. We wish to know what you want us to do with this old sprint submission and review. Since all the code is changed, it would be redundant to fix this old code just to replace with the new. We have taken a look at each comment during the sprint, and made sure we complied with them in the new code.

File: docs/legal/footer.md [hunk 5]
```diff
@@ -0,0 +1,2 @@
+This will be the footer, probably of the format "Copyright © Rugged Software / Hestia (current year)", along with whatever contact information we want to include.
```
[the remark this answers is not in the document]
Comment (PapaOscar90) [reply docs/legal/footer.md]: Roman has researched legal information for the next sprint. 👌

File: site/main.py [hunk 6]
```diff
@@ -5,35 +5,46 @@
...
 
 app = Flask(__name__,
-    static_url_path='',
-    static_folder='templates/static')
+            static_url_path='',
+            static_folder='static')
 
+# Timeout for requests.
+
+TIMEOUT = 1.00
```
[the remark this answers is not in the document]
Comment (PapaOscar90) [reply site/main.py]: We will be implementing a change for this is in the future. But we are still investigating a-sync. For the time being we can factor this out into a separate file.

File: site/main.py [hunk 7]
```diff
@@ -5,35 +5,46 @@
...
 
 # Create endpoints
+
```
[the remark this answers is not in the document]
Comment (PapaOscar90) [reply site/main.py]: 🤦‍♂️ We are ensuring future code does not have unnecessary white space.

File: site/main.py [hunk 8]
```diff
@@ -5,35 +5,46 @@
...
     return render_template('settings.html')
 
+
 @app.route('/login.html')
 def login():
     return render_template('login.html')
 
+
 @app.route('/request', methods=['POST'])
 def apiRequestHandler():
```
[the remark this answers is not in the document]
Comment (PapaOscar90) [reply site/main.py]: 👍  We have conformed to snake_case for our new code.

File: site/static/scripts/ui/domInteraction.js (hunk 1 above)
Comment (PapaOscar90) [reply site/static/scripts/ui/domInteraction.js]: 👍 The old GUI is entirely replaced this sprint. We are ensuring our GUI is easy to navigate and conforms to the topics from the lecture.

File: site/static/scripts/ui/domInteraction.js (hunk 2 above)
Comment (PapaOscar90) [reply site/static/scripts/ui/domInteraction.js]: 🤦‍♂️ 👍 We will leave To-Do out of the code for now on.

File: site/static/scripts/ui/domInteraction.js (hunk 4 above)
Comment (PapaOscar90) [reply site/static/scripts/ui/domInteraction.js]: 👍 This is replaced in the new frontend.

**Comment by Feikoritsema** [comment]
Thanks for the good clarification, and I will merge for now, but do indeed keep all the comments (and general clean code ) in mind for the next sprint(s).

**Pull request merged**